pdc/apps/release/serializers.py

```diff
@@ -151,17 +151,6 @@ def update(self, instance, validated_data):
         if hasattr(instance, 'pk'):
             # reload to make sure changes in mapping are reflected
             obj = Release.objects.get(pk=obj.pk)
-        # By default, PUT does not erase optional field if not specified. This
-        # loops over all optional fields and resets them manually.
-        if not self.partial:
-            for field_name, field in self.fields.iteritems():
-                if not field.read_only and field_name not in validated_data:
-                    attr = field.source or field_name
-                    try:
-                        if hasattr(obj, attr):
-                            setattr(obj, attr, None)
-                    except ValueError:
-                        pass
```

Comment:
Do you mean in 1.11 the put option have already erased the optional fields?
If so, I hope we can create an issue to track this change and let QE guys to test it.

Reply:
I have no idea why it's not needed now but the tests pass (and there are specific tests for this behavior).

Reply:
Just tested the behavior:
```bash
> pdc_client -r releases/acbu-8271 -x PUT -d '{"release_type": "ga", "short": "acbu", "version": "8271", "name": "adcjysacbu", "allow_buildroot_push": "true"}'
{
    ...
    "allow_buildroot_push": true, 
    ...
}

> pdc_client -r releases/acbu-8271 -x PUT -d '{"release_type": "ga", "short": "acbu", "version": "8271", "name": "adcjysacbu"}'                                
{
    ...
    "allow_buildroot_push": false, 
    ...
}
```

Reply:
Yes, I saw the tests, and also test this codes in 1.8 for allowed_buildroot_push, we removing these codes it will keep the previous value when I put without a value.
But adding these codes it will set default without a value in the parameters. so removing the codes in 1.11 get the same results.
Here from my view it's ok to remove these codes, not sure if we should ask QE to test this, hope there is no issue in ct's scripts :)

Reply:
lol, you and me did the same test